# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/qt6base/hotfix-clone2.patch.ia64
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- qtbase-everywhere-src-6.10.1/src/3rdparty/forkfd/forkfd_linux.c.vanilla	2025-12-07 14:49:47.091251405 +0100
+++ qtbase-everywhere-src-6.10.1/src/3rdparty/forkfd/forkfd_linux.c	2025-12-07 14:45:54.143493280 +0100
@@ -53,6 +53,12 @@
 
 #define SYSTEM_FORKFD_CAN_VFORK
 
+#ifdef __ia64
+int __clone2(typeof(int (void *)) *fn,
+             void *stack_base, size_t stack_size,
+             int flags, void *arg, ...);
+#endif
+
 // in forkfd.c
 static int convertForkfdWaitFlagsToWaitFlags(int ffdoptions);
 static void convertStatusToForkfdInfo(int status, struct forkfd_info *info);
@@ -169,7 +175,12 @@
     }
     *system = 1;
 
+#ifdef __ia64
+    pid = __clone2(childFn, childStack, sizeof(childStack), cloneflags, token, &pidfd, NULL, NULL);
+#else
     pid = clone(childFn, childStack + sizeof(childStack), cloneflags, token, &pidfd, NULL, NULL);
+#endif
+
     if (pid < 0)
         return pid;
     if (ppid)
